### PR TITLE
updated by adding wait logic & dynamic gke version

### PR DESCRIPTION
**Pull request by edwardmedia**

### CODE REVIEW

**Comment by chrislovecnm** [comment]
We are hitting an issue with CI.  We are getting the following error:

```
Wait for the BACKEND k8s-ig--e8eb079bdb87e679 being deleted.
Wait for the BACKEND k8s-ig--e8eb079bdb87e679 being deleted.
Wait for the BACKEND k8s-ig--e8eb079bdb87e679 being deleted.
Wait for the BACKEND k8s-ig--e8eb079bdb87e679 being deleted.
Wait for the BACKEND k8s-ig--e8eb079bdb87e679 being deleted.
Retry limit exceeded; exiting...
Makefile:98: recipe for target 'tf-destroy' failed
```

Are you addressing this?

/cc @pydevops

**Comment by edwardmedia** [comment]
Yes. There was no teardown stage in the Jenkinsfile. Just added

File: Jenkinsfile [hunk 1]
```diff
@@ -129,6 +129,14 @@ spec:
       }
     }
 
+    // Teardown the resources created during the Create stage
+    stage('Teardown') {
+      steps {
+        container('k8s-node') {
+             sh 'USER=jenkins gcloud compute ssh jenkins-deploy-dev-infra@gke-tutorial-bastion --command \'scripts/teardown.sh\''
```
Comment (chrislovecnm) [inline Jenkinsfile]: We should not need the USER= with the new Jenkins file ... let me merge that and you can rebase.

```suggestion
             sh 'gcloud compute ssh jenkins-deploy-dev-infra@gke-tutorial-bastion --command \'scripts/teardown.sh\''
```

File: waitfor_svc_deleted.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+# this code waits for the backend services to be deleted
+LIST='hello-root|hello-user|hello-override|armored-hello-user|armored-hello-denied'
+REGION="$(gcloud config get-value compute/region)"
+
+STR="$(gcloud compute backend-services list | awk '{print $1}' | awk '!/NAME/')"
+
+ARR=$(echo $STR | tr " " "\n")
+
+for ITEM in $ARR
```
Comment (chrislovecnm) [inline waitfor_svc_deleted.sh]: We need some code level comments please

**Comment by edwardmedia** [comment]
test this please

**Comment by edwardmedia** [comment]
Rerun

**Comment by edwardmedia** [comment]
Can we rerun this

File: README.md [hunk 3]
```diff
@@ -83,7 +83,7 @@ In order to use the code in this demo you will need access to the following tool
 * Terraform v0.11.7
 * gcloud v206.0.0
 * kubectl v1.10.4
-* Kubernetes Engine v1.10
+* Kubernetes Engine (latest validMasterVersion)
```
Comment (chrislovecnm) [inline README.md]: Not sure if people will know what this means.
```suggestion
* Kubernetes Engine - latest released version
```

File: terraform/modules/firewall/variables.tf [hunk 4]
```diff
@@ -28,3 +28,8 @@ variable "net_tags" {
   description = "tags for the firewall"
   type        = "list"
 }
+
+variable "execution_id" {
+  description = "The random number for each execution"
```
Comment (chrislovecnm) [inline terraform/modules/firewall/variables.tf]:
```suggestion
  description = "A random string used as as suffix for resources."
```

Comment (chrislovecnm) [reply terraform/modules/firewall/variables.tf]: Shouldn't we be using one random string for the entire project?  Would make deleting resources A LOT simpler.

File: terraform/modules/instance/variables.tf [hunk 5]
```diff
@@ -39,6 +39,11 @@ variable "zone" {
   type        = "string"
 }
 
+variable "execution_id" {
```
Comment (chrislovecnm) [inline terraform/modules/instance/variables.tf]: please use one id for the entire project.

File: waitfor_svc_deleted.sh [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+REGION="$(gcloud config get-value compute/region)"
+
+# obtain the list of backend services within the application
+STR="$(gcloud compute backend-services list | awk '{print $1}' | awk '!/NAME/')"
+
+ARR=$(echo $STR | tr " " "\n")
+
+# Pick one service name, $ITEM, in order to find the BACKEND service name $NAME
+for ITEM in $ARR
+do
```
Comment (chrislovecnm) [inline waitfor_svc_deleted.sh]: we need code level documentation throughout this file please.

File: terraform/modules/firewall/variables.tf (hunk 4 above)
Comment (edwardmedia) [reply terraform/modules/firewall/variables.tf]: That execution_id is used across the entire project.

**Pull request merged**